social_core/backends/discourse.py

```diff
@@ -0,0 +1,86 @@
+from .base import BaseAuth
+from ..exceptions import AuthException, AuthTokenError
+import hmac
+from base64 import b64encode, b64decode
+from hashlib import sha256
+import urllib
+from ..utils import parse_qs
+import time
+
+
+class DiscourseAuth(BaseAuth):
+    name = "discourse"
+    EXTRA_DATA = ["username", "name", "avatar_url"]
+
+    def auth_url(self):
+        """Return redirect url"""
+        returnUrl = self.redirect_uri
+        nonce = self.strategy.random_string(64)
+        self.add_nonce(nonce)
+
+        payload = "nonce=" + nonce + "&return_sso_url=" + returnUrl
+        base64Payload = b64encode(payload)
+        payloadSignature = hmac.new(
+            self.setting("SOCIAL_AUTH_DISCOURSE_AUTH_SECRET"), base64Payload, sha256
```

Comment:
You can drop the `SOCIAL_AUTH_DISCOURSE_` from the setting name, the `setting()` method will add the prefix automatically.